src/Microsoft.PowerShell.Commands.Utility/commands/utility/Csv.cs

```diff
@@ -27,85 +29,59 @@ internal CSVHelper(char delimiter)
         /// String to be parsed.
         /// </param>
         internal Collection<string> ParseCsv(string csv)
-        {
-            Collection<string> result = new Collection<string>();
-            string tempString = string.Empty;
+        {        
             csv = csv.Trim();
+            Collection<string> result = new Collection<string>();
             if (csv.Length == 0 || csv[0] == '#')
             {
                 return result;
             }
 
-            bool inQuote = false;
-            for (int i = 0; i < csv.Length; i++)
+            var reader = new StringReader(csv);
+            StringBuilder wordBuffer = new StringBuilder();
+
+            while (reader.Peek() != -1) 
             {
-                char c = csv[i];
-                if (c == Delimiter)
+                char nextChar = (char)reader.Read();
+
+                // if next character was delimiter or we are at the end, add string to result and clear builder
+                // else if next character was quote, perform reading untill next quote and add it to builder
+                // else read and add it to builder
+                if (nextChar == Delimiter) 
                 {
-                    if (!inQuote)
-                    {
-                        result.Add(tempString);
-                        tempString = string.Empty;
-                    }
-                    else
-                    {
-                        tempString += c;
-                    }
-                }
-                else
+                    result.Add(wordBuffer.ToString());
+                    wordBuffer.Clear();
+                } 
+                else if (nextChar == '"') 
                 {
-                    switch (c)
+                    bool betweenQuotes = true;
```

Comment:
We already use `inQuotes` in tokenizer so maybe it is just better.

Reply:
At the moment I'm not sure which you currently think is best

Reply:
Hungarian notation warning can be ignored for now. 

Dongbo has a PR pending that is disabling some of the CodeFactor rules the team don't agree with. 🙂

Reply:
then `inQuotes` it is!